ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -62,8 +67,11 @@ openMempool registry chainDB cfg = do
 
 -- | Internal state in the mempool
 data InternalState blk = IS {
-     -- | Transactions currently in the mempool
-      isTxs :: TxSeq (GenTx blk)
+      -- | Transactions currently in the mempool
+      isTxs :: TxSeq (GenTxId blk, GenTx blk)
```

Comment:
So I guess the idea is that we keep both, since getting the txid is expensive since we do not rely on the `GenTx` caching its own txid, And we decided we wanted to return the txid in query functions.

We should document that caching design here. It's a bit subtle, since in other parts of the code we assume that getting a hash style Id is cheap because it's cached.